main.nf

```diff
@@ -0,0 +1,533 @@
+      --gzi                         Path to bgzip index format (.gzi) produced by faidx
+      *Pass all five files above to skip the fasta preprocessing step
+
+      Options:
+      -profile                      Configuration profile to use. Can use multiple (comma separated)
+                                    Available: standard, conda, docker, singularity, awsbatch, test
+      --exome                       For exome bam files
+      --rgid                        Bam file read group line id incase its needed (default = 4)
+      --rglb                        Bam file read group line library incase its needed (default = 'lib1')
+      --rgpl                        Bam file read group line platform incase its needed (default = 'illumina')
+      --rgpu                        Bam file read group line platform unit incase its needed (default = 'unit1')
+      --rgsm                        Bam file read group line sample incase its needed (default = 20)
+
+    Other options:
+      --outdir                      The output directory where the results will be saved (default = results)
+      --email                       Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits
+      -name                         Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
+      --help                        Bring up this help message
+
+    AWSBatch options:
+      --awsqueue                    The AWSBatch JobQueue that needs to be set when running on AWSBatch
+      --awsregion                   The AWS Region for your AWS Batch job to run on
+    """.stripIndent()
+}
+
+/*
+ * SET UP CONFIGURATION VARIABLES
+ */
+
+// Show help emssage
+if (params.help){
+    helpMessage()
+    exit 0
+}
+
+//set model for call variants either whole genome or exome
+model= params.exome ? 'wes' : 'wgs'
+
+//if user inputs fasta set their reference files otherwise use hg19 as default
+if(params.fasta){
+  fasta = params.fasta
+  fai = params.fai ? params.fai : false
+  fastagz = params.fai ? params.fastagz : false
+  gzfai = params.fai ? params.gzfai : false
+  gzi = params.fai ? params.gzi : false
+  bed = params.bed
+} else {
+  params.genome = 'hg19'
+}
+
+//set genome options
+if (params.genome){
+  fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+  fai = params.genome ? params.genomes[ params.genome ].fai ?: false : false
+  fastagz = params.genome ? params.genomes[ params.genome ].fastagz ?: false : false
+  gzfai = params.genome ? params.genomes[ params.genome ].gzfai ?: false : false
+  gzi = params.genome ? params.genomes[ params.genome ].gzi ?: false : false
+  bed = params.genome ? params.genomes[ params.genome ].bed ?: false : false
+}
+
+//setup fasta channels for preprocessing fasta files
+(fastaCh, fastaCh1, fastaCh2, fastaCh3, fastaCh4) = Channel.fromPath(fasta).into(5)
```

Comment:
fastaCh, fastaCh1, fastaCh2, fastaCh3, fastaCh4 could be more explicit
Like: fastaCh, fastaToIndexCh, fastaToGzCh...